### PR TITLE
Improve readability of source code

### PR DESCRIPTION
**Pull request by fb55**
I recently ran into #1606 (see https://github.com/inikulin/parse5/pull/467). I spent a couple of hours trying to understand the source code of TypeDoc, and made small changes whenever it took me longer to understand a piece of code than it should have. The commits should be helpful as a description of why things were changed.

I've targeted the `beta` branch — hope this is where commits should be added.

### CODE REVIEW

File: src/lib/utils/sort.ts [hunk 1]
```diff
@@ -26,21 +26,18 @@ const sorts: Record<
     (a: DeclarationReflection, b: DeclarationReflection) => boolean
 > = {
     "source-order"(a, b) {
-        const aSymbol = a.project.getSymbolFromReflection(a);
```
Comment (Gerrit0) [inline src/lib/utils/sort.ts:29]: This change is not ok. Sorting should be able to use source order even if `--disableSources` is passed.

File: src/lib/validation/documentation.ts [hunk 2]
```diff
@@ -14,23 +12,15 @@ export function validateDocumentation(
     );
 
     for (const ref of project.getReflectionsByKind(kinds)) {
-        const symbol = project.getSymbolFromReflection(ref);
```
Comment (Gerrit0) [inline src/lib/validation/documentation.ts:17]: This change is not ok, this needs to work even if `--disableSources` is passed.

File: src/lib/utils/entry-point.ts [hunk 3]
```diff
@@ -139,7 +139,7 @@ function getEntryPointsForPaths(
 
     entryLoop: for (const fileOrDir of inputFiles.map(normalizePath)) {
         const toCheck = [fileOrDir];
-        if (!/\.[tj]sx?/.test(fileOrDir)) {
+        if (!/\.[tj]sx?$/.test(fileOrDir)) {
```
Comment (Gerrit0) [inline src/lib/utils/entry-point.ts:142]: Good catch!

**Review by Gerrit0 — CHANGES_REQUESTED** [review]
I like most of the changes, thanks! A few comments...

**Comment by Gerrit0** [comment]
Ignore the CI failure for visual regression, not caused by your changes.

File: src/lib/converter/plugins/SourcePlugin.ts [hunk 4]
```diff
@@ -176,10 +176,9 @@ export class SourcePlugin extends ConverterComponent {
         const project = context.project;
         const home = project.directory;
         project.files.forEach((file) => {
-            const reflections: DeclarationReflection[] = [];
-            file.reflections.forEach((reflection) => {
-                reflections.push(reflection);
-            });
+            const reflections: DeclarationReflection[] = file.reflections.map(
```
[the remark this answers is not in the document]
Comment (fb55) [reply src/lib/converter/plugins/SourcePlugin.ts:179]: `.map` already creates a new array :)

[the remark this answers is not in the document]
Comment (Gerrit0) [reply src/lib/converter/plugins/SourcePlugin.ts:179]: Yep, with slice we don't need the map that looks like it'll do something though ;)

[the remark this answers is not in the document]
Comment (fb55) [reply src/lib/converter/plugins/SourcePlugin.ts:179]: Good catch, didn't look at the part below.

File: src/lib/utils/sort.ts (hunk 1 above)
Comment (fb55) [reply src/lib/utils/sort.ts]: Thanks for flagging — oversaw this.

File: src/lib/converter/factories/index-signature.ts [hunk 5]
```diff
@@ -46,10 +46,6 @@ export function convertIndexSignature(context: Context, symbol: ts.Symbol) {
         context.registerReflection(index, indexSymbol);
         context.scope.indexSignature = index;
 
-        context.trigger(
-            ConverterEvents.CREATE_SIGNATURE,
-            index,
-            indexDeclaration
-        );
+        context.trigger(ConverterEvents.CREATE_SIGNATURE, index);
```
[the remark this answers is not in the document]
Comment (fb55) [reply src/lib/converter/factories/index-signature.ts]: Reverted the change.

**Pull request merged**